src/core/server/plugins/types.ts

```diff
@@ -153,6 +153,12 @@ export interface PluginManifest {
    * Specifies whether plugin includes some server-side specific functionality.
    */
   readonly server: boolean;
+
+  /**
+   * Specifies directory names that can be imported by other plugins sharing
+   * the same instance of the @kbn/optimizer.
+   */
+  readonly extraPublicDirs?: string[];
```

Comment:
I'm surprised it doesn't update https://github.com/elastic/kibana/blob/master/docs/development/core/server/kibana-plugin-core-server.pluginmanifest.md
You might need to run `yarn docs:acceptApiChanges --docs`